src/odata-to-abstract-sql.ts

```diff
@@ -513,20 +533,29 @@ export class OData2AbstractSQL {
 						abstractSql: bindVarSelectQuery,
 					};
 				} else {
-					unionResource.definition = {
-						...convertToModernDefinition(unionResource.definition),
-					};
-					if (unionResource.definition.abstractSql[0] !== 'SelectQuery') {
+					const rewrittenBindVars: ModernDefinition['binds'] = [];
+					const definition: Definition = (unionResource.definition =
+						this.rewriteDefinition(
+							unionResource.definition,
+							rewrittenBindVars,
+							0,
+						));
+					definition.binds = rewrittenBindVars;
+
+					if (
+						definition.abstractSql[0] !== 'SelectQuery' &&
+						definition.abstractSql[0] !== 'Table'
+					) {
 						throw new Error(
-							'Only select query definitions supported for inserts',
+							'Only SelectQuery or Resource definitions supported for inserts',
 						);
 					}
-
+					const tableName = unionResource.modifyName ?? unionResource.name;
 					const isTable = (part: any): part is TableNode =>
-						isTableNode(part) && part[1] === unionResource.name;
+						isTableNode(part) && part[1] === tableName;
 
-					if (isTable(unionResource.definition.abstractSql)) {
-						unionResource.definition.abstractSql = bindVarSelectQuery;
+					if (isTable(definition.abstractSql)) {
```

Comment:
Nope, because `isTable` isn't really `isTable`, it's more `isMatchingTable` - I'll fix this to make it clearer because it also confused me initially when you asked

Reply:
I still think it was dead code before, b/c abstract-sql-complier's isTableNode is
```
export const isTableNode = (n: AbstractSqlType): n is TableNode =>
	n[0] === 'Table';
```

and previously here the code was doing
```
if (unionResource.definition.abstractSql[0] !== 'SelectQuery') { throw... }
```
which means that the `if (isTable(unionResource.definition.abstractSql))` that we have right after, should be always false, since `isTableNode` checks whether `unionResource.definition.abstractSql[0] === 'Table'` while we have already confirmed it's `SelectQuery`. Right? :thinking: 

See: https://github.com/balena-io-modules/abstract-sql-compiler/blob/df382c95382ef8c5fa91d71665909f28084a4517/src/AbstractSQLCompiler.ts#L438-L439
See: https://github.com/balena-io-modules/odata-to-abstract-sql/blob/78d37262e05d932fbc20ff4265bf9118028447a6/src/odata-to-abstract-sql.ts#L519-L528

Anyway doesn't really matter :)

Reply:
Ah, yeah that part of the `if` check would have been dead indeed, I was looking at other uses